sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/physical/partitioning.scala

```diff
@@ -519,6 +519,35 @@ case class BroadcastPartitioning(mode: BroadcastMode) extends Partitioning {
   }
 }
 
+/**
+ * Represents a partitioning use partition value map to partition.
+ */
+case class PartitionValueMapPartitioning(
```

Comment:
Could we reuse `KeyGroupedPartitioning` instead of creating a new `Partitioning`? they look mostly similar, except `valueMap` which I think can be computed using `partitionValues`? The `PartitionValueMapShuffleSpec` also look similar to `KeyGroupedShuffleSpec`